core/utils/thread_control.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"context"
+	"sync"
+)
+
+var _ ThreadControl = &threadControl{}
+
+// ThreadControl is a helper for managing a group of goroutines.
+type ThreadControl interface {
+	// Go starts a goroutine and tracks the lifetime of the goroutine.
+	Go(fn func(context.Context))
+	// Close cancels the goroutines and waits for all of them to exit.
+	Close()
+}
+
+func NewThreadControl() *threadControl {
+	tc := &threadControl{
+		stop: make(chan struct{}),
+	}
+
+	return tc
+}
+
+type threadControl struct {
+	threadsWG sync.WaitGroup
+	stop      StopChan
```

Comment:
The purpose of the WG is fairly clear since it's used idiomatically here. 

I am having some trouble understanding the purpose of `stop` - could you explain and add some documentation?

Reply:
You can find comments above in the interface, I'll add some details in the PR description.

In general I'm using the StopChan to signal all the running goroutines to exit.
You can find usage examples in https://github.com/smartcontractkit/chainlink/pull/10535